command/repo.go

```diff
@@ -63,6 +82,146 @@ func repoClone(cmd *cobra.Command, args []string) error {
 	return utils.PrepareCmd(cloneCmd).Run()
 }
 
+func repoCreate(cmd *cobra.Command, args []string) error {
+	projectDir, projectDirErr := git.ToplevelDir()
+
+	orgName := ""
+	teamSlug, err := cmd.Flags().GetString("team")
+	if err != nil {
+		return err
+	}
+
+	var name string
+	if len(args) > 0 {
+		name = args[0]
+		if strings.Contains(name, "/") {
+			newRepo := ghrepo.FromFullName(name)
+			orgName = newRepo.RepoOwner()
+			name = newRepo.RepoName()
+		}
+	} else {
+		if projectDirErr != nil {
+			return projectDirErr
+		}
+		name = path.Base(projectDir)
+	}
+
+	isPublic, err := cmd.Flags().GetBool("public")
+	if err != nil {
+		return err
+	}
+	hasIssuesEnabled, err := cmd.Flags().GetBool("enable-issues")
+	if err != nil {
+		return err
+	}
+	hasWikiEnabled, err := cmd.Flags().GetBool("enable-wiki")
+	if err != nil {
+		return err
+	}
+	description, err := cmd.Flags().GetString("description")
+	if err != nil {
+		return err
+	}
+	homepage, err := cmd.Flags().GetString("homepage")
+	if err != nil {
+		return err
+	}
+
+	// TODO: move this into constant within `api`
+	visibility := "PRIVATE"
+	if isPublic {
+		visibility = "PUBLIC"
+	}
+
+	input := api.RepoCreateInput{
+		Name:             name,
+		Visibility:       visibility,
+		OwnerID:          orgName,
+		TeamID:           teamSlug,
+		Description:      description,
+		Homepage:         homepage,
+		HasIssuesEnabled: hasIssuesEnabled,
+		HasWikiEnabled:   hasWikiEnabled,
+	}
+
+	ctx := contextForCommand(cmd)
+	client, err := apiClientForContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	repo, err := api.RepoCreate(client, input)
+	if err != nil {
+		return err
+	}
+
+	out := cmd.OutOrStdout()
+	greenCheck := utils.Green("✓")
+	isTTY := false
+	if outFile, isFile := out.(*os.File); isFile {
+		isTTY = utils.IsTerminal(outFile)
+		if isTTY {
+			// FIXME: duplicates colorableOut
+			out = utils.NewColorable(outFile)
```

Comment:
I do want to see this cleaned up and better abstracted but am ok with that being follow-up work